boa/environment.py

```diff
@@ -212,6 +212,7 @@ def deploy(
         override_address: Optional[_AddressType] = None,
         # the calling vyper contract
         contract: Any = None,
+        contract_name: Optional[str] = None,  # TODO: This isn't used
```

Comment:
We probably don't need this, we can call `contract.contract_name`

Reply:
If someone passes `contract_name = xxx` to the deploy of a non `NetworkEnv` `boa.env`, this would revert. So I think we do need this to make it compatible with the `NetworkEnv` object

Reply:
i think it would be an error to pass `contract_name = xxx` to env.deploy().